contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +171,44 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+* `--discovery-token` If set, this is a token that is created in the new cluster that is used for signing discovery bootstrap.  If not set, it defaults to the value from `--token`.  If set to the empty string, there is no token created.
+* `--discovery-token-ttl` Similar TTL for token.
+* `--bootstrap-auth-token` (not strictly part of this proposal) If set, this token is set into the cluster to be used for doing authentication for TLS bootstrap.  If set to the empty string there is no token created.
+* `--boostrap-auth-token-ttl`
+
+### `kubeadm join` flags
+
+* `--token` This sets the token for both discovery and bootstrap auth.
+* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL.  This will support `http` and `https+insecure` but it we will warn if this is not `https`.  This also supports `username:password@host` syntax for doing HTTP auth.
+* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
+* `--discovery-token-url` This is the URL to hit to get the ConfigMap that contains the kubeconfig and signatures.  This defaults to `https+insecure://<api-server>/api/v1/namespaces/kube-public/configmaps/cluster-info`.
+
+Only one of `--discovery-url`, `--discovery-file` or `--discovery-token` can be set.  If more than one is set then an error is surfaced and `kubeadm join` exits.  Setting `--token` counts as setting `--discovery-token`.
+
+### `kubeadm token` commands
+
+`kubeadm` provides a set of utilities for manipulating token secrets in a running server.
+
+* `kubeadm token create` Creates a token server side.  With no options this'll create a token that is used for discovery and TLS bootstrap.
+  * `--token` The actual token value (in `id.secret` form) to write in.  If unset, a random value is generated.
+  * `--usages` A list of usages.  Defaults to `signing,authentication`
+  * `--ttl` The TTL for this token.
+* `kubeadm token delete <token-id>|<token-id>.<token-secret>`
+  * Users can either just specify the id or the secret.  This will delete the token if it exists.
+* `kubeadm token list`
+  * List tokens in a table form listing out the `token-id.token-secrert`, the TTL, the absolute expiration time and the usages.
```

Comment:
nit: token-id.token-secret
So you mean this should be like this with both the full expiration time and the time that's left from `Now()`?
```
TOKEN                                   TTL    EXPIRATION                       USAGES
abcdef.0123456789abcdef      8h      01.01.2018T00:00:00.000   signing,authentication
```

or just 
```
TOKEN                                   TTL    USAGES
abcdef.0123456789abcdef      8h     signing,authentication
```